### PR TITLE
eval_model uses bsz when counting num_examples, batchsort loaded from saved opt

### PR DESCRIPTION
**Pull request by alexholdenmiller**

### CODE REVIEW

**Comment by alexholdenmiller** [comment]
@emilydinan a change to override opt!

File: parlai/core/agents.py [hunk 1]
```diff
@@ -300,6 +300,8 @@ def load_agent_module(opt):
                 new_opt[k] = v
         new_opt['model_file'] = model_file
         model_class = get_agent_module(new_opt['model'])
+        if 'batch_sort' in new_opt:
+            opt['batch_sort'] = new_opt['batch_sort']
```
Comment (emilydinan) [inline parlai/core/agents.py]: is this backwards? the model is created with `new_opt` so why do we need to change `opt`?

Comment (uralik) [reply parlai/core/agents.py]: exactly, but the batch sorting is happening not in the model, but we want it to use setting which was use during the model training

Comment (jaseweston) [reply parlai/core/agents.py]: this is  weird and ugly to some special case flags here like that..  we need to find a nicer way of doing what you want i think.. e.g. why only batch_sort
and as emily said, i dont think it's expected this function is changing its input

Comment (jaseweston) [reply parlai/core/agents.py]: can u please unmerge this, and work out a nicer way

**Pull request merged**